### ISSUE TITLE
AtomicParsley exit code 6 

### ISSUE THREAD

**Issue by DarkJamie** [griff/metaz#217]
I have found that when the "Director", Producer", and/or "Screenwriter" fields have too many entries, this error occurs. It took me a long time to figure that out, but once I did, I haven't had this error anymore. Just truncate these fields. Maybe even the"Actors" field as well. Guess there's a character limit on said.

**Comment by griff** [comment]
There must be something else going on. I just tried to reproduce this issue locally by putting a bonkers amount of actors (460), directors (950) and screenwriters (460) and I don't get this error.

You are right that the 4 fields are related as they are internally converted to a plist and written to the same field in the mp4 file but it must be something other than just the amount.

What macOS version are you on and what version of MetaZ are you using?

**Comment by DarkJamie** [comment]
Primary machine
2010 MacPro
MacOS 20.13.6
MetaZ : Version 1.0.beta-49 (19.06.19.11.d5e7d11)



Secondary machine
2013 Black MacPro
MetaZ : Version 1.0.beta-49 (19.06.19.11.d5e7d11)
MacOS: 10.14.6


It could be that there are offensive characters in the long strings of names. Too many names to see them all. I just know that the error has only occurred for me when any one of those fields is loaded with data. In these cases, I usually select the first 5 to 10 names and delete the rest. That always solves the problem.

**Comment by griff** [comment]
You are right that it is probably be some characters that it doesn't like. If you come across the problem again could I get you to copy/paste the entire values of those long fields into a comment on this issue?

**Comment by DarkJamie** [comment]
Sure. If I happen across it, I'll post all field data.

**Comment by DarkJamie** [comment]
Just got the error again on the first listing for Titanic (1997). The "actors" field is extremely long, and thus throwing the code 6. 

Kate Winslet, Leonardo DiCaprio, Billy Zane, Kathy Bates, Frances Fisher, Gloria Stuart, Bill Paxton, Bernard Hill, David Warner, Victor Garber, Jonathan Hyde, Suzy Amis, Lewis Abernathy, Nicholas Cascone, Danny Nucci, Jason Barry, Lew Palter, Eric Braeden, Bernard Fox, Ewan Stewart, Ioan Gruffudd, Jonny Phillips, Edward Fletcher, Scott G. Anderson, Martin East, Gregory Cooke, Alexandrea Owens, Seth Adkins, Michael Ensign, Anatoly M. Sagalevitch, Martin Hub, Mark Lindsay Chapman, Richard Graham, Paul Brightwell, Ron Donachie, Charlotte Chatton, Fannie Brett, Jenette Goldstein, Camilla Overbye Roos, Linda Kerns, Amy Gaipa, Martin Jarvis, Rosalind Ayres, Rochelle Rose, Jonathan Evans-Jones, Rocky Taylor, Craig Kelly, Liam Tuohy, Simon Crane, James Lancaster, Elsa Raven, Reece P. Thompson III, Laramie Landis, Mark Rafael Truitt, John Walcutt, Terry Forrestal, Derek Lea, Richard Ashton, Sean Nepita, Brendan Connolly, David Cronnelly, Garth Wilton, Richard Fox, Nick Meaney, Kevin Owers, Mark Capri, Marc Cass, Paul Herbert, Emmett James, Chris Byrne, Oliver Page, James Garrett, Erik Holland, Jari Kinnunen, Anders Falk, Barry Dennen, Vern Urich, Rebecca Klingler, Tricia O'Neil, Kathleen S. Dunn, Romeo Francis, Mandana Marino, Van Ling, Bjørn Olsen, Dan Pettersson, Shay Duffin, Greg Ellis, Diana Morgan, Kris Andersson, Bobbie Bates, Aaron James Cash, Anne Fletcher, Edmond Alan Forsyth, Andie Hicks, Scott Hislop, Stan Mazin, Lisa Ratzin, Julene Renee, Brian Walsh, Alexandra Boyd, Mike Butters, James Cameron, Bruno Campolo, Kevin De La Noy, Tony Kenny, Sean Lawlor, Don Lynch, Johnny Martin, Ryan McClurkin, Meghan McLeod, Mike O'Neal, Phil Parlapiano, Steven Quale, Olivia Rosewood, John Slade, R. Gern Trowbridge

**Comment by griff** [comment]
I could finally reproduce this myself. It seems to be a memory bug in AtomicParsley. What triggers it I am not totally sure about. Taking a file with no metadata and applying the metadata from Titanic triggered it but only the first time. After that the bug wasn't triggered on that file again.

Until I get a new version of AtomicParsley compiled (that hopefully has a fix for this) or I rewrite the entire metadata writing process there sadly isn't much I can do to prevent this bug.

**Comment by DarkJamie** [comment]
Hopefully people read this and see that they can still get the tags to work, with a little end user action. Once I discovered that it happens, It's easy enough to work around it.

**Comment by till213** [comment]
Hello,

I just stumbled over this exact same issue for the first time. It happened for the movie "Plane" (with Gerard Butler):

https://www.imdb.com/title/tt5884796/

The hints about the "Actors" field were spot on. It turns out that the following actor name causes trouble ("exit code 6"):

- Enele Ma'afu Tauteoli

Removing the ' character and hence changing the actor name to:

- Enele Maafu Tauteoli

worked around this issue. Perhaps the ' character could be escaped, such that Parsley is able to parse it?


This is with:
- MetaZ 1.0.3
- macOS 15.0 "Sequoia"
- Intel iMac 27" 2020

**Comment by LDMartin1959** [comment]
This is still an issue. Is this on the "to be fixed really, really soon" list?

**Comment by Beameranged** [comment]
> Removing the ' character and hence changing the actor name

Such a helpful comment. For me it was O'Connor — also in the _Actors_ field — and removing the apostrophe indeed solved it.
